x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/SearchableSnapshotAction.java

```diff
@@ -328,6 +351,55 @@ public List<Step> toSteps(Client client, String phase, StepKey nextStepKey, XPac
         );
 
         // If a new snapshot is needed, these steps are executed
+        // If the index has replicas, we need to clone the index first with 0 replicas and perform the force-merge on that index.
+        // That avoids us having to force-merge the replica shards too, which is a waste, as the snapshot will only be taken from the
+        // primary shards. If the index already has 0 replicas, we can skip the clone steps.
+        BranchingStep conditionalSkipCloneStep = new BranchingStep(
+            conditionalSkipCloneKey,
+            readOnlyKey,
+            forceMergeStepKey,
+            (index, project) -> {
+                IndexMetadata indexMetadata = project.index(index);
+                assert indexMetadata != null : "index " + index.getName() + " must exist in the cluster state";
+                return indexMetadata.getNumberOfReplicas() == 0;
+            }
+        );
+        ReadOnlyStep readOnlyStep = new ReadOnlyStep(readOnlyKey, cleanupClonedIndexKey, client, false);
+        // If a previous step created a clone index but the action did not complete, we need to clean up the old clone index.
+        CleanupGeneratedIndexStep cleanupClonedIndexStep = new CleanupGeneratedIndexStep(
+            cleanupClonedIndexKey,
+            generateCloneIndexNameKey,
+            client,
+            FORCE_MERGE_INDEX_NAME_SUPPLIER
+        );
+        GenerateUniqueIndexNameStep generateCloneIndexNameStep = new GenerateUniqueIndexNameStep(
+            generateCloneIndexNameKey,
+            cloneIndexKey,
+            FORCE_MERGE_INDEX_PREFIX,
+            (generatedIndexName, lifecycleStateBuilder) -> lifecycleStateBuilder.setForceMergeIndexName(generatedIndexName)
+        );
+        // Clone the index with 0 replicas.
+        ResizeIndexStep cloneIndexStep = new ResizeIndexStep(
+            cloneIndexKey,
+            waitForClonedIndexGreenKey,
+            client,
+            ResizeType.CLONE,
+            FORCE_MERGE_INDEX_NAME_SUPPLIER,
+            CLONE_SETTINGS_SUPPLIER,
+            null
+        );
+        // Wait for the cloned index to be green before proceeding with the force-merge. We wrap this with a
+        // ClusterStateWaitUntilThresholdStep to avoid waiting forever if the index cannot be started for some reason.
+        // On timeout, ILM will move back to the cleanup step, remove the cloned index, and retry the clone.
+        ClusterStateWaitUntilThresholdStep waitForClonedIndexGreenStep = new ClusterStateWaitUntilThresholdStep(
+            new WaitForIndexColorStep(
+                waitForClonedIndexGreenKey,
+                forceMergeStepKey,
+                ClusterHealthStatus.GREEN,
+                FORCE_MERGE_INDEX_NAME_SUPPLIER
+            ),
+            cleanupClonedIndexKey
```

Comment:
Hm, I thought I copied the approach of the `ShrinkAction` here by going back to the cleanup step if the threshold/timeout is passed. But it looks like that's not the case: 
https://github.com/elastic/elasticsearch/blob/1ff6608260d0844d2fc20aa5c3aa05b67b669c51/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ShrinkAction.java#L267-L270
The `ShrinkAction` just goes back to `SetSingleNodeAllocateStep`. I'm inclined to think my current approach is safer, but I'm also a fan of consistency. Anyone else have any thoughts?